src/main/webapp/data-visualization.js

```diff
@@ -349,3 +330,13 @@ function toggleDataTable() {
     document.getElementById('toggle-data-btn').innerText = 'Display raw data';
   }
 }
+
+// Display link to data.census.gov table for the table the displayed
+// data is from.
+function displayLinkToCensusTable(tableString, year) {
+  const tablePrefix = tableString[0] === 'S' ? 'SPP' : 'DP';
```

Comment:
There's actually three kinds of tables: the ones that start with DP, the ones that start with SPP, and the ones that start with S (unfortunately). It looks like this doesn't take into account the ones that start with S.
Also, those are just the three kinds we're using right now, so potentially we might have to change this code if we added a fourth kind. Is there a way to do these calculations in the Java backend, so that the javascript doesn't have to know anything about SPP vs DP vs S, and it's all kept in one place when we need to change it?

Reply:
Thanks for letting me know, moved link generation all to backend